### ISSUE TITLE
Crash on very large PNG image in iTerm2 sequence

### ISSUE THREAD

**Issue by klamonte** [wez/wezterm#1031]
Found another crash: pass a very large (100000 x 100000) base64-encoded PNG as iTerm2 image, wezterm crashes:

```
memory allocation of 10000000000 bytes failed
Aborted
```

Here is one such sequence: https://gitlab.com/klamonte/term-crash-tests/-/blob/main/tests/image/big-damn-iterm2.out

**Comment by wez** [comment]
Which version are you using? I could have sworn that I fixed a bug like this already, but maybe that was sixel

**Comment by klamonte** [comment]
You fixed the analagous one for sixel. I'm perverse. :-)

I'm using a nightly from a couple weeks back:

```
$ wezterm --version
wezterm 20210804-162508-e3acbd59
```

**Comment by klamonte** [comment]
BTW creating a large png was not the easiest thing.  Googling around I found instructions for using libvips:

```
vips black black.png 100000 100000 --bands 1
```

**Comment by klamonte** [comment]
I do like this test because several protocols basically say "take this blob and hand it off to an image decoding library".  What happens when that library is given bad data?  Will it play nice and fail to allocate, or will it try to get 9-37GB of memory (depending on 8-32 bits/pixel)?

**Comment by klamonte** [comment]
@wez Same crash with kitty.

Attached is a test input.

[kitty_large_png.zip](https://github.com/wez/wezterm/files/8003933/kitty_large_png.zip)

(There are other errors in kitty protocol.  I will file those separately a bit later.)

**Comment by wez** [comment]
Thanks; should be fixed by 3e2f5263bc847be2d6b6a21a2d38fa65c5b8ffbd
